test/integration/integration_test.py

```diff
@@ -0,0 +1,85 @@
+from contextlib import contextmanager
+from marquez_client.marquez import MarquezClient
+from marquez_codegen_client.rest import ApiException
+from time import sleep
+from urllib3.exceptions import (MaxRetryError,
+                                NewConnectionError,
+                                ReadTimeoutError)
+
+import pytest
+import requests
+import subprocess
+import sys
+
+MARQUEZ_HOST = "marquez"
+MARQUEZ_PORT = "5000"
+
+
+def test_bad_host():
+    assert wait_for_marquez()
+    c = MarquezClient(host="bad-host", port=MARQUEZ_PORT)
+    with pytest.raises(MaxRetryError) as e:
+        c.get_namespace_info("no_connection")
+    assert isinstance(e.value.reason, NewConnectionError)
+
+
+def test_bad_port():
+    assert wait_for_marquez()
+    c = MarquezClient(host=MARQUEZ_HOST, port="6000")
+    with pytest.raises(MaxRetryError) as e:
+        c.get_namespace_info("no_connection")
+    assert isinstance(e.value.reason, NewConnectionError)
+
+
+def test_timeout():
+    assert wait_for_marquez()
+    c = MarquezClient(host=MARQUEZ_HOST, port=MARQUEZ_PORT, timeout=1)
+
+    with broken_network():
+        expected_namespace = "timeout_test"
+        with pytest.raises(MaxRetryError) as e:
+            c.get_namespace_info(expected_namespace)
+        assert isinstance(e.value.reason, ReadTimeoutError)
+
+
+def test_namespace_not_found():
+    assert wait_for_marquez()
+    c = MarquezClient(host=MARQUEZ_HOST, port=MARQUEZ_PORT)
+
+    expected_namespace = "not_found"
+    with pytest.raises(ApiException):
+        c.get_namespace_info(expected_namespace)
+
+
+@contextmanager
+def broken_network():
+    iptables_drop_packets(True)
+    yield
+    iptables_drop_packets(False)
+
+
+def iptables_drop_packets(drop):
+    subprocess.run(["iptables",
+                    "-A" if drop else "-D", "INPUT",
+                    "-p", "tcp",
+                    "--tcp-flags", "PSH", "PSH",
+                    "--sport", MARQUEZ_PORT,
+                    "-j", "DROP"])
+
+
+def wait_for_marquez(retries=10):
```

Comment:
I'd strongly recommend leveraging existing capabilities for timeouts and retries. There is also support for the amount of time to wait between retries, status codes that should be retried (e.g. 5xx), among other things.

```
from requests.adapters import HTTPAdapter
s = requests.Session()
s.mount('http://stackoverflow.com', HTTPAdapter(max_retries=5))
```
For each request verb, e.g. `request.get`, there is a parameter you can add to set the timeout: 
```
import requests
from requests.exceptions import Timeout

try:
    response = requests.get('https://api.github.com', timeout=1)
```

Reply:
done